modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -682,6 +682,16 @@ else if (value instanceof Object[]) {
 					document.addDate(name.concat("_date"), dateValues);
 				}
 			}
+			else if (type.equals(DDMFormFieldTypeConstants.RICH_TEXT)) {
+				List<String> richTextValues = new ArrayList<>(
+					valuesString.length);
+
+				for (String valueString : valuesString) {
+					richTextValues.add(_htmlParser.extractText(valueString));
+				}
```

Comment:
I also removed the html check from here since it was just adding up more code and it didn't make a difference from what I was testing it.